lib/ansible/executor/task_queue_manager.py

```diff
@@ -316,6 +351,7 @@ def has_dead_workers(self):
                 defunct = True
         return defunct
 
+    @lock_decorator(attr='_callback_lock')
```

Comment:
Due to the implementation here, I don't actually need an instance attr, I could pass the lock in directly.